src/datasets.c

```diff
@@ -357,6 +357,8 @@ static int DatasetLoadString(Dataset *set)
         }
     }
 
+    // TODO figure out how to reset memcap after load is finished, a reset function? a macro?
```

Comment:
Maybe just a small util func like `THashConsolidateMemcap` which takes the memuse and sets memcap to that value. You can call it directly after the load completes.